### ISSUE TITLE
Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.

### ISSUE THREAD

**Issue by muriloventuroso** [Baseflow/flutter_cached_network_image#721]
```
Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
../…/image_provider/multi_image_stream_completer.dart:152
- 'SchedulerBinding' is from 'package:flutter[/src/scheduler/binding.dart]()' ('.[./../../fvm/versions/beta/packages/flutter/lib/src/scheduler/binding.dart]()').
package:flutter/…/scheduler/binding.dart:1
    SchedulerBinding.instance?.scheduleFrameCallback(_handleAppFrame);
                     ^
```

Flutter 2.12.0-4.1.pre

**Comment by muhammadmateen027** [comment]
I'm facing below warnings since I upgraded the flutter version to `2.13.0-0.1.pre`.

**Warning**: 
```dart
./../flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/multi_image_stream_completer.dart:152:22: Warning: Operand of null-aware
operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../flutter/packages/flutter/lib/src/scheduler/binding.dart').
    SchedulerBinding.instance?.scheduleFrameCallback(_handleAppFrame);
                     ^
../../flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/cached_network_image_provider.dart:109:29: Warning: Operand of null-aware
operation '?.' has type 'PaintingBinding' which excludes null.
 - 'PaintingBinding' is from 'package:flutter/src/painting/binding.dart' ('../../flutter/packages/flutter/lib/src/painting/binding.dart').
      () => PaintingBinding.instance?.imageCache?.evict(key),
```

**Flutter version**:
```dart
Flutter 2.13.0-0.1.pre • channel unknown • unknown source
Framework • revision 13a2fb10b8 (8 days ago) • 2022-04-12 15:34:25 -0500
Engine • revision 499984f99c
Tools • Dart 2.17.0 (build 2.17.0-266.1.beta) • DevTools 2.12.1
```

**Comment by AliPunjabi** [comment]
@muhammadmateen027 nope, this is not solving my issue

**Comment by xXUlquiScXx** [comment]
i'm facing the same problem in Flutter 3.0.0

**Comment by AliPunjabi** [comment]
> > @muhammadmateen027 nope, this is not solving my issue
> 
> I mentioned, I am facing the same issue but `Not Solution`.
I solved the problem but forget how, I think by changing flutter channel from master to stable or stable to master

**Comment by cesarkmelo** [comment]
The same here after upgrading to 3.0.0 in Android

**Comment by andreclinio** [comment]
Same problem after flutter 3.0

**Comment by gaddlord** [comment]
Started appearing on every run after upgrade to 3.0.0

/C:/Users/USERNAME/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/flutter_swiper_null_safety-1.0.2/lib/src/custom_layout.dart:32:20: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../../flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
    WidgetsBinding.instance!.addPostFrameCallback(_getSize);
                   ^

**Comment by isanecat** [comment]
Same problem after flutter 3.0

**Comment by AliPunjabi** [comment]
Guys try to change the flutter channel from master to stable or stable to master, it will work

**Comment by Franion03** [comment]
Same problem after flutter 3.0

**Comment by Abdu1269** [comment]
same here after upgrading to flutter 3.0

**Comment by Antunees** [comment]
Same problem after flutter 3.0

**Comment by heshaShawky** [comment]
same here too after 3.0

**Comment by lucasrueda** [comment]
I created the following PR
https://github.com/Baseflow/flutter_cached_network_image/pull/737

If you want to test it, replace this in your pubspec.yaml

```
  cached_network_image:
    git:
      url: https://github.com/lucasrueda/flutter_cached_network_image.git
      path: cached_network_image
      ref: develop
```

**Comment by nazaawa** [comment]
same probleme with flutter 3.0

**Comment by thitlwincoder** [comment]
Same problem after flutter 3.0

**Comment by md-siam** [comment]
Same problem after upgrading to Flutter 3.0
I upgraded all the third-party packages, but the problem remains. 😮‍💨
![Screen Shot 2022-05-14 at 3 44 01 PM](https://user-images.githubusercontent.com/21225918/168420460-40444b65-2797-4860-84bb-eebf116bd2c9.png)

**Comment by Sheikh-Yawar** [comment]
same problem after upgrading to flutter 3.0

**Comment by AliPunjabi** [comment]
check which version of flutter u have and check the channel

check flutter version: flutter --version

check flutter channel: flutter channel

to change from master to stable channel or vice versa: flutter channel stable

then do flutter upgrade.

**Comment by AliPunjabi** [comment]
> i'm facing the same problem in Flutter 3.0.0, with an iOS device
> 
> ![image](https://user-images.githubusercontent.com/6279443/168129566-a3d3e379-2691-416b-9ba4-79a73a47ccd3.png)
> 
> ![image](https://user-images.githubusercontent.com/6279443/168130304-05149b7a-0cc4-4a44-b2b6-376724491ab0.png)

change to flutter channel master and upgrade flutter

**Comment by AliPunjabi** [comment]
It should be a non-fatal warning. Either ignore it or temporarily switch from the master branch to the latest Flutter stable release until the problematic package is updated.
Just change the channel.

**Comment by md-siam** [comment]
**SOLUTION:**

1. upgrade all the thirst-party packages in pubspec.yaml
2. flutter clean
3. delete pubspec.lock file
4. flutter pub get

Done!!! ✌🏻

**Comment by tebalink** [comment]
../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flare_flutter-3.0.2/lib/flare_render_box.dart:167:26: Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../development/flutter/packages/flutter/lib/src/scheduler/binding.dart').
        SchedulerBinding.instance?.cancelFrameCallbackWithId(_frameCallbackID);
                         ^
../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flare_flutter-3.0.2/lib/flare_render_box.dart:170:28: Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../development/flutter/packages/flutter/lib/src/scheduler/binding.dart').
          SchedulerBinding.instance?.scheduleFrameCallback(_beginFrame) ?? -1;
                           ^
../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flare_flutter-3.0.2/lib/flare_render_box.dart:272:26: Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../development/flutter/packages/flutter/lib/src/scheduler/binding.dart').
        SchedulerBinding.instance?.cancelFrameCallbackWithId(_frameCallbackID);
                         ^
../../development/flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/multi_image_stream_completer.dart:152:22: Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('../../development/flutter/packages/flutter/lib/src/scheduler/binding.dart').
    SchedulerBinding.instance?.scheduleFrameCallback(_handleAppFrame);
                     ^
../../development/flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/cached_network_image_provider.dart:109:29: Warning: Operand of null-aware operation '?.' has type 'PaintingBinding' which excludes null.
 - 'PaintingBinding' is from 'package:flutter/src/painting/binding.dart' ('../../development/flutter/packages/flutter/lib/src/painting/binding.dart').
      () => PaintingBinding.instance?.imageCache?.evict(key),

**Comment by frankdavid-addae** [comment]
> **SOLUTION:**
> 
>     1. upgrade all the thirst-party packages in pubspec.yaml (optional)
> 
>     2. flutter clean
> 
>     3. delete pubspec.lock file
> 
>     4. flutter pub get
> 
> 
> Done!!! ✌🏻

I did all of these yet the issue still persists.

**Comment by mmasood22** [comment]
i am also facing the same issue after upgrading to flutter 3.0.0

**Comment by lucasrueda** [comment]
@mmasood22 @frankdavid-addae @Sheikh-Yawar Please test this:

https://github.com/Baseflow/flutter_cached_network_image/issues/721#issuecomment-1126569540

**Comment by Ahmed-Mechrir** [comment]
Same problem after flutter 3.0.0

**Comment by renefloor** [comment]
Update released as cached_network_image: ^3.2.1

**Comment by md-siam** [comment]
[renefloor](https://github.com/renefloor) many developers are not upgrading their packages. 😒

One of my projects is completely un-usable on Flutter 3.0

**Comment by micky113** [comment]
i am not sure if i am right..   I got this  error after connecting firebase and making some changes to gradle.
so next day, after uninstalling flutter, android studio (including sdk) and reinstalling.. i am not getting this error but i still havent reconnected firebase as of now.

**Comment by murad-alm** [comment]
I have the same issue  on Flutter 3.0.1

**Comment by micky113** [comment]
> i am not sure if i am right.. I got this error after connecting firebase and making some changes to gradle. so next day, after uninstalling flutter, android studio (including sdk) and reinstalling.. i am not getting this error but i still havent reconnected firebase as of now.

So, now after again connecting firebase, i realised that i was doing some mistakes while copying files to the gradle.
Now after following the instructions on firebase, i have no error.

**Comment by jobic10** [comment]
Perhaps, downgrade a bit till when this issue has been fixed.
However, it won't stop you from building.

**Comment by isanecat** [comment]
There is also a discussion on stackoverflow possibly useful for the case. 
https://stackoverflow.com/questions/72210806/warning-operand-of-null-aware-operation-has-type-schedulerbinding-which-e

**Comment by MohammedAlabadla** [comment]
#solution
Update released as cached_network_image: ^3.2.1
remove ! or ? from the code
flutter clean
flutter pub get
flutter run
..
have a good code 🌹
